pkg/pillar/types/contenttreetypes.go

```diff
@@ -134,7 +136,7 @@ func (status ContentTreeStatus) Key() string {
 
 // ResolveKey will return the key of resolver config/status
 func (status ContentTreeStatus) ResolveKey() string {
-	return fmt.Sprintf("%s+%s+%v", status.DatastoreID.String(),
+	return fmt.Sprintf("%s+%s+%v", status.DatastoreIDsList[0].String(),
```

Comment:
Makes sense using the same concatenation of string UUIDs I suggested for the logs here as well.

Reply:
For the logging case the main question is do we limit the number of datastores in the list? 
For this particular case: in this PR I do not solve the SHA resolving for OCI registry, that's why it is left as is.

Reply:
But for this particular case I can `strings.Join()` all the uuids, seems to be `resolvekey()` should work.

Reply:
I think using strings.Join makes sense.
The main open question is whether (now or later) we want to have the resolveConfig/resolveStatus be for a single datastore with failure recovery loop in volumemgr, or have the loop inside the resolver. In the former case then having a single datastore UUID in resolveConfig makes sense; in the latter a strings.Join.

Given that the downloader does the loop for the downloads it might make sense to have it do the loop for the resolve step as well, but I haven't looked at the code recently.